### ISSUE TITLE
Provide something like len() for GzEncoder and GzDecoder

### ISSUE THREAD

**Issue by kolbma** [alexcrichton/flate2-rs#170]
Hi,
would be nice to know the size of the extracted data, although there is an overflow after 4GB.
Or is it available and have not found it?!

Thanks

**Comment by alexcrichton** [comment]
Thanks for the report! I don't think this is something that can be implemented though? Deflate streams don't know how big they're going to become, such information would have to be sent through a different channel alongside the stream to learn that

**Comment by oyvindln** [comment]
If you mean the amount of bytes extracted **so far**, the underlying encoder does know. The counter may be 32-bit though. It's used when decompressing gzip-wrapped streams as they have a 32-bit length check value at the end. It's not available from the read/write structs though.

Knowing the extracted length before extracting everything is not really possible though.

**Comment by kolbma** [comment]
Yes think of the last 4 bytes. I've implemented this for me to get the value and do some proof. I thought it would be nice to have this in-build in flate2.
For sure this is not possible if the input is a stream and you have not get the last 4 bytes. So there would be the requirement of some error handling. 
But this should be possible:
```
extern crate flate2;

use std::io::prelude::*;
use flate2::read::GzDecoder;

fn main() {
    let mut d = GzDecoder::new("...".as_bytes());
    let mut s = String::new();
    d.read_to_string(&mut s).unwrap();
    let sizefooter: usize = d.get_size_of_footer().unwrap();  // <=====
    println!("{}", s);
}
```
